tests/scripts/all.sh

```diff
@@ -697,6 +697,45 @@ component_test_full_cmake_gcc_asan () {
     if_build_succeeded tests/compat.sh
 }
 
+component_test_zlib_make() {
+    msg "build: zlib enabled, make"
+    scripts/config.py set MBEDTLS_ZLIB_SUPPORT
+    make ZLIB=1 CFLAGS='-Werror -O1'
+
+    msg "test: main suites (zlib, make)"
+    make test
+
+    msg "test: ssl-opt.sh (zlib, make)"
+    if_build_succeeded tests/ssl-opt.sh
+}
+support_test_zlib_make () {
+    base=support_test_zlib_$$
+    cat <<'EOF' > ${base}.c
+#include "zlib.h"
+int main(void) { return 0; }
+EOF
+    gcc -o ${base}.exe ${base}.c -lz 2>/dev/null
+    ret=$?
```

Comment:
`ret=$?` relies on `set -e` being inactive. Given that `support_test_zlib_make` is only ever called through
```
if ! support_$component; then …
```
this works, because `set -e` is deactivated in the `if` clause. But it's fragile. That's why I suggested a more complex version whose behavior is independent of whether `set -e` is in effect.